PBTutorialManager/PBTutorialManager/lib/TutorialManager.swift

```diff
@@ -61,9 +62,31 @@ open class TutorialManager: NSObject {
             return
         }
         
+        // Under some rare corner cases a the view target may not be visible when the tutorial is
+        // fired, even if the tutorial is started from viewDidAppear(). One case where this is known
+        // to happen is for buttons on a tool bar in the detail view of a splitViewController. For
+        // some strange reason these buttons visibly appear about 0.5 seconds after the rest of the
+        // view controller if the master view also has a tool bar.
         if let currentTarget = targets.first {
-            currentTarget.onShow?()
-            showTarget(currentTarget)
+            if currentTarget.view?.window != nil {
+                timeOutCount = 0
+                currentTarget.onShow?()
+                showTarget(currentTarget)
+            } else if timeOutCount < 20 {
```

Comment:
Is this an arbitrary constant ? And if Yes why ?

Reply:
Yes it's an arbitrary timeout of 2 seconds. I've expanded the comment to explain the rational.